src/linear_feedback_controller.cpp

```diff
@@ -58,33 +63,45 @@ const Eigen::VectorXd& LinearFeedbackController::compute_control(
     first_control_received_time_ = time;
   }
 
+  if (remove_gravity_compensation_effort) {
+    robot_model_builder_->construct_robot_state(sensor, robot_configuration_,
+                                                robot_velocity_);
+
+    // NOTE: .tail() is used to remove the freeflyer components
+    tau_gravity_ =
+        pinocchio::rnea(robot_model_builder_->get_model(),
+                        robot_model_builder_->get_data(), robot_configuration_,
+                        robot_velocity_null_, robot_velocity_null_)
+            .tail(tau_init_.size());
+  }
+
   if (!first_control_received_time_initialized) {
     control_ =
         pd_controller_.compute_control(sensor_js.position, sensor_js.velocity);
+
+    if (remove_gravity_compensation_effort) {
+      control_ -= tau_init_;
+    }
   } else if (during_switch) {
     double weight = ((time - first_control_received_time_).count()) /
                     params_.pd_to_lf_transition_duration.count();
     weight = std::clamp(weight, 0.0, 1.0);
-    const Eigen::VectorXd& pd_ctrl =
+    control_tmp_ =
         pd_controller_.compute_control(sensor_js.position, sensor_js.velocity);
-    const Eigen::VectorXd& lf_ctrl =
-        lf_controller_.compute_control(sensor, control);
+    control_ = lf_controller_.compute_control(sensor, control);
+
+    if (remove_gravity_compensation_effort) {
+      control_tmp_ -= tau_init_;
+      control_ -= tau_gravity_;
+    }
 
-    control_ = (1.0 - weight) * pd_ctrl + weight * lf_ctrl;
+    control_ = (1.0 - weight) * control_tmp_ + weight * control_;
```

Comment:
Please keep the ctrl_pd and control_lf variable. because when you do:
```
control_ = (1.0 - weight) * control_tmp_ + weight * control_;
```
You have aliases management under the hood. Here it's not crazy inefficient because it's a few doubles. But in general using Eigen one would do:
```suggestion
    control_.no_alias() = (1.0 - weight) * control_pd_ + weight * control_lf_;
```

Reply:
If you introduce a `control_lf_`, you don't need to write control_.no_alias() right? 
If I understand, you propose 2 alternatives:

1) Introduce a new attribute `control_lf_` and write:   
`control_ = (1.0 - weight) * control_pd_ + weight * control_lf_;`

2) Make the alias management explicit:
`control_.no_alias() = (1.0 - weight) * control_pd_ + weight * control_;`
?

Reply:
The no_alias forces Eigen to not check if the alias management is necessary or not.
It's a small trick that is not absolutely needed but that was useful to raise my case.

Option 1 is enough for me I think. And clearer in the code in my humble opinion.

Reply:
Do we need the .no_alias() at line 98? As the multiplication is scalar only, the whole expression is element-wise, so Eigen would not assume an aliasing.

Reply:
You don't need it again it's a rule of thumb of Eigen to be aware of these topics.